crates/ruma-common/src/push.rs

```diff
@@ -296,10 +296,16 @@ impl Ruleset {
 
         if event.get_str("sender").is_some_and(|sender| sender == context.user_id) {
             // no need to look at the rules if the event was by the user themselves
-            None
-        } else {
-            self.iter().find(|rule| rule.applies(&event, context))
+            return None;
+        }
+
+        for rule in self.iter() {
```

Comment:
Do we not have an `IntoIterator` impl for `&Ruleset`?

```suggestion
        for rule in &self {
```